lib/subscription-connection.js

```diff
@@ -207,12 +209,20 @@ module.exports = class SubscriptionConnection {
     this.subscriptionContexts.delete(data.id)
   }
 
-  handleConnectionClose () {
+  async handleConnectionClose () {
```

Comment:
who is calling this function? Are you sure they are using `await` on it? I think a better approach is add a `.catch()` handler instead.

Reply:
Yeah, I agree that using .catch() would be better, but initially I wasn't entirely too sure how to use `.catch()` on a callback that could be potentially both synchronous and asynchronous. After a bit of searching, I ended up using `Promise.resolve().then(() => cb()).catch((e) => { log(e) })`, but I'm not sure if this is the best approach.

Let me know if you know of a better way to handle this!